src/js/game/items/boolean_item.js

```diff
@@ -48,15 +50,19 @@ export class BooleanItem extends BaseItem {
      * @param {DrawParameters} parameters
      */
     drawItemCenteredImpl(x, y, parameters, diameter = globalConfig.defaultItemDiameter) {
-        let sprite;
-        if (this.value) {
-            sprite = Loader.getSprite("sprites/wires/boolean_true.png");
-        } else {
-            sprite = Loader.getSprite("sprites/wires/boolean_false.png");
-        }
+        const value = Boolean(this.value).toString();
+        const sprite = Loader.getSprite(`sprites/wires/boolean_${value}.png`);
         sprite.drawCachedCentered(parameters, x, y, diameter);
     }
 }
 
 export const BOOL_FALSE_SINGLETON = new BooleanItem(0);
 export const BOOL_TRUE_SINGLETON = new BooleanItem(1);
+
+/**
+ * @param {unknown} item
+ * @returns {item is BooleanItem}
+ **/
+export const isBooleanItem = item => {
+    return item instanceof BooleanItem;
```

Comment:
Don't use instanceof, that's too slow. Instead use `getItemType() === "boolean"`

Reply:
```
export const isBooleanItem = item => {
    return item instanceof BooleanItem;
};

const isItemType = item => {
    return item && item.getItemType() === "boolean";
};

import { COLOR_ITEM_SINGLETONS } from "./color_item";

window.benchmark1 = () => {
    console.time('instanceof');

    for (let i = 0; i < 100000; i += 1) {
        isBooleanItem(BOOL_FALSE_SINGLETON);
        isBooleanItem(COLOR_ITEM_SINGLETONS.red);
    }

    console.timeEnd('instanceof');
};

window.benchmark2 = () => {
    console.time('itemType');

    for (let i = 0; i < 100000; i += 1) {
        isItemType(BOOL_FALSE_SINGLETON);
        isItemType(COLOR_ITEM_SINGLETONS.red);
    }

    console.timeEnd('itemType');
};

for (let i = 0; i < 10; i += 1) {
    window.benchmark1();
    window.benchmark2();
}
```

We get results like this:
```
instanceof: 3.31494140625ms
itemType: 4.391357421875ms
instanceof: 7.876953125ms
itemType: 7.22509765625ms
instanceof: 0.050048828125ms
itemType: 0.0419921875ms
instanceof: 0.042236328125ms
itemType: 0.038818359375ms
instanceof: 0.0419921875ms
itemType: 0.038818359375ms
instanceof: 0.041015625ms
itemType: 0.0400390625ms
instanceof: 0.0419921875ms
itemType: 0.0458984375ms
instanceof: 0.0478515625ms
itemType: 0.044921875ms
instanceof: 0.05712890625ms
itemType: 0.046142578125ms
instanceof: 0.048095703125ms
itemType: 0.046142578125ms
```
 The difference is negligible, especially when v8 optimizes the function.

The advantage of the `instanceof` check is that we can pass anything in there, while if we rely on the class method, we have to be sure that we do a null check first.

Reply:
Interesting! is it also the case on firefox and older browsers?

Reply:
On Firefox where the game greets you with this message:

![shapez-chrome](https://user-images.githubusercontent.com/20630/90531376-f3dc0d80-e1b0-11ea-939b-fe3734cc1d2e.png)

```
instanceof: 2ms - timer ended
itemType: 3ms - timer ended
instanceof: 3ms - timer ended
itemType: 3ms - timer ended
instanceof: 2ms - timer ended
itemType: 0ms - timer ended
instanceof: 2ms - timer ended
itemType: 1ms - timer ended
instanceof: 1ms - timer ended
itemType: 0ms - timer ended
instanceof: 2ms - timer ended
itemType: 0ms - timer ended
instanceof: 2ms - timer ended
itemType: 0ms - timer ended
instanceof: 3ms - timer ended
itemType: 1ms - timer ended
instanceof: 2ms - timer ended
itemType: 0ms - timer ended
instanceof: 2ms - timer ended
itemType: 1ms - timer ended
```